src/embind/embind.js

```diff
@@ -2015,9 +2021,8 @@ var LibraryEmbind = {
                 if (arguments.length !== argCount - 1) {
                     throwBindingError(humanName + ' called with ' + arguments.length + ' arguments, expected ' + (argCount-1));
                 }
-                var destructors = [];
-                var args = new Array(argCount);
-                args[0] = rawConstructor;
+                destructors.length = 0;
+                args.length = argCount;
```

Comment:
Should we assert `argCount != 0`? Otherwise we would lose the `rawConstructor` arg. I'm not sure if that's fine. Alternatively: `args.length = Math.max(1, argCount);`

Reply:
Added the assert.

Reply:
(There are several places that assume that it's positive.)